### PR TITLE
feat(chat): clarify chat reference

### PR DESCRIPTION
**Pull request by EthanWaite**
Made some minor changes to the chat reference:
* Separated out each chat method's arguments and required permissions, and then added them to methods that didn't have any docs there before.
* Clarified timeout duration & how giveaways work.
* Fixed a couple of incorrect method names.
* Added missing `originatingChannel` to poll events.
* Fixed links for methods with a colon.

### CODE REVIEW

File: src/reference/chat/index.pug [hunk 1]
```diff
@@ -120,8 +120,16 @@ block reference
                 td Associated event data - may be of any type, specific to the event.
     h1(id='chat__methods') Methods
     for data, method in chat.methods
-        h3(id=`chat__methods__${method}`)= method
+        h3(id=`chat__methods__${method.replace(/\W/g, '_')}`)= method
```
Comment (ProbablePrime) [inline src/reference/chat/index.pug]: What's the purpose of this?

Comment (EthanWaite) [inline src/reference/chat/index.pug]: Changes the id for `vote:choose` to `vote_choose` so it works with scrollspy

File: src/reference/chat/data.json [hunk 2]
```diff
@@ -1,7 +1,12 @@
 {
   "methods": {
     "auth": {
-      "description": "Authenticate as an active user in a specified channel. Arguments are `channelId`, `userId`, `key`. The channelId is the id of the channel you wish to join. The userId is the id of the user you are joining as. The key is retrieved from a request to our REST API as explained in the [Connection](#chat__connection) section. \n\r You can connect anonymously by just supplying the `channelId` as an argument. If you do this you will not be able to send messages or participate in chat. This can be useful for creating chat overlays.",
```
Comment (SimonSchick) [inline src/reference/chat/data.json]: \n should suffice, no?

Comment (EthanWaite) [inline src/reference/chat/data.json]: ooh, yeah it should be `\n\n`, thanks!

File: src/reference/chat/index.pug (hunk 1 above)
Comment (ProbablePrime) [inline src/reference/chat/index.pug]: aahhh :) ok!

**Comment by connor4312** [comment]
Can you screenshot what these new argument descriptions look like?

**Comment by EthanWaite** [comment]
Yep, just went for a numbered list (feedback is welcome 😄) --

screenshot

**Comment by EthanWaite** [comment]
Also just noticed the response there still says "duration", whoops

**Comment by connor4312** [comment]
Looks good! For accessibilty purposes, however, the `Arguments` heading should be a true `<h*>` heading, no just `<strong>`

**Comment by EthanWaite** [comment]
Okay, I'll switch it :)

**Comment by connor4312** [comment]
🚢 🇮🇹

**Pull request merged**